New Azure Pipelines does shallow fetch by default

### Problem description

New Azure Pipelines do shallow fetch by default which make GitVersion to fail since all commit history is not available.

### Verbose logs

```text
Not available
```


### How to reproduce

Run the current pipeline in a newly created Azure Pipelines pipeline.

### Expected behavior

Should not do shallow fetch.

### Current behavior

Does shallow fetch.

### Suggested solution

Add the following to [azure-pipelines.yml](https://github.com/gaelcolas/Sampler/blob/main/azure-pipelines.yml):

```yaml
variables:
  Agent.Source.Git.ShallowFetchDepth: 0
```

### Operating system the target node is running

```text
n/a
```


### PowerShell version and build the target node is running

```text
n/a
```


### Module version used

```text
n/a
```

@johlju, Is it the reasons why I'm having this issue when deploying the pipeline
(it was my first attempt on this project today)

```
Starting: Calculate ModuleVersion (GitVersion)
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.212.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/home/vsts/work/_temp/0f944673-58bd-4ea6-8125-ce009ad98b07.ps1'
You can invoke the tool using the following command: dotnet-gitversion
Tool 'gitversion.tool' (version '5.12.0') was successfully installed.
ConvertFrom-Json: /home/vsts/work/_temp/0f944673-58bd-4ea6-8125-ce009ad98b07.ps1:4
Line |
   4 |  $gitVersionObject = dotnet-gitversion | ConvertFrom-Json
     |                                          ~~~~~~~~~~~~~~~~
     | Conversion from JSON failed with error: Error parsing Infinity value.
     | Path '', line 1, position 1.

##[error]PowerShell exited with code '1'.
Finishing: Calculate ModuleVersion (GitVersion)
```
My `GitVersion.yml` is the following:

``` YAML
mode: ContinuousDelivery
next-version: 0.0.1
major-version-bump-message: '(breaking\schange|breaking|major)\b'
minor-version-bump-message: '(adds?|features?|minor)\b'
patch-version-bump-message: '\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
assembly-informational-format: '{NuGetVersionV2}+Sha.{Sha}.Date.{CommitDate}'
branches:
  master:
    tag: preview
    regex: ^main$
  pull-request:
    tag: PR
  feature:
    tag: useBranchName
    increment: Minor
    regex: f(eature(s)?)?[\/-]
    source-branches: ['master']
  hotfix:
    tag: fix
    increment: Patch
    regex: (hot)?fix(es)?[\/-]
    source-branches: ['master']

ignore:
  sha: []
  merge-message-formats: {}

feature:
  tag: useBranchName
  increment: Minor
  regex: f(eature(s)?)?[/-]
  source-branches: ['master']

hotfix:
  tag: fix
  increment: Patch
  regex: (hot)?fix(es)?[/-]
  source-branches: ['master']

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Fix Azure Pipeline bug to resolve errors and delays during the build process. Shallow fetch has been disabled to ensure complete repository cloning. Fixes [#424](https://github.com/gaelcolas/Sampler/issues/424)
+
 ## [0.116.4] - 2023-04-06
 
 ### Fixed

--- a/Sampler/Templates/Sampler/azure-pipelines.yml.template
+++ b/Sampler/Templates/Sampler/azure-pipelines.yml.template
@@ -16,6 +16,7 @@ variables:
   buildArtifactName: output
   testResultFolderName: testResults
   defaultBranch: <%= ${PLASTER_PARAM_MainGitBranch} %>
+  Agent.Source.Git.ShallowFetchDepth: 0 # override ShallowFetchDepth
 
 stages:
   - stage: Build

--- a/Sampler/Templates/Sampler/azure-pipelines_dsccommunity.yml.template
+++ b/Sampler/Templates/Sampler/azure-pipelines_dsccommunity.yml.template
@@ -18,6 +18,7 @@ variables:
   testArtifactName: testResults
   sourceFolderName: source
   defaultBranch: <%= ${PLASTER_PARAM_MainGitBranch} %>
+  Agent.Source.Git.ShallowFetchDepth: 0 # override ShallowFetchDepth
 
 stages:
   - stage: Build

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -16,6 +16,7 @@ variables:
   buildArtifactName: output
   testResultFolderName: testResults
   defaultBranch: main
+  Agent.Source.Git.ShallowFetchDepth: 0 # override ShallowFetchDepth
 
 stages:
   - stage: Build

